### PR TITLE
Update teacher and student sign-up flow tests

### PR DESCRIPTION
**Pull request by TurnerRiley**
As per [this request](https://github.com/code-dot-org/code-dot-org/pull/60883#discussion_r1747661468) in [this PR](https://github.com/code-dot-org/code-dot-org/pull/60883) (which ensured that the state question was only shown to students in the U.S.), this PR adds tests for the student flow to check that the submit button is disabled/enabled when the required fields are empty/completed.

In addition, I simplified the clearing of `sessionStorage` in both the teacher and student test files.

## Links
PR comment: [here](https://github.com/code-dot-org/code-dot-org/pull/60883#discussion_r1747661468)

## Testing story
Successfully ran new tests locally.

### CODE REVIEW

File: apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx [hunk 1]
```diff
@@ -190,143 +180,139 @@ describe('FinishStudentAccount', () => {
...
-  it('finish student signup button starts disabled', () => {
-    renderDefault();
-
-    const finishSignUpButton = screen.getByRole('button', {
-      name: locale.go_to_my_account(),
-    });
-    expect(finishSignUpButton.getAttribute('aria-disabled')).toBe('true');
-  });
-
-  it('finish student signup button is disabled until required fields are entered without errors', () => {
```
Comment (TurnerRiley) [inline apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:285]: I removed this general test that does not specifically test one particular field, instead these checks were split up across the tests above

File: apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx [hunk 2]
```diff
@@ -190,143 +180,139 @@ describe('FinishStudentAccount', () => {
...
+    expect(finishSignUpButton.getAttribute('aria-disabled')).toBe(null);
 
     // Clear state
     fireEvent.change(parentEmailInput, {target: {value: ''}});
 
-    // Error shows with empty state
-    screen.getByText(locale.email_error_message());
-  });
-
-  it('finish student signup button starts disabled', () => {
```
Comment (TurnerRiley) [inline apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:276]: This was an accidental duplicate of a test above

**Review by bethanyaconnor — COMMENTED** [review]
Love the new/refactored tests! Not to be nitpicky, but I don't see the same checks for `does not render state question if user is not detected in the U.S.` (the original test we were discussing). Can we use a similar pattern for that test?

**Comment by TurnerRiley** [comment]
> Love the new/refactored tests! Not to be nitpicky, but I don't see the same checks for `does not render state question if user is not detected in the U.S.` (the original test we were discussing). Can we use a similar pattern for that test?

Ahh shoot you're so right sorry! I got lost in format updating I forgot the core of it.

**Review by bethanyaconnor — APPROVED** [review]
🎉

**Pull request merged**